src/common/Collision/Management/VMapManager.cpp

```diff
@@ -33,14 +33,21 @@ using G3D::Vector3;
 
 namespace VMAP
 {
-    VMapManager2::VMapManager2()
+    VMapManager::VMapManager() : iEnableLineOfSightCalc(true), iEnableHeightCalc(true)
     {
         GetLiquidFlagsPtr = &GetLiquidFlagsDummy;
         IsVMAPDisabledForPtr = &IsVMAPDisabledForDummy;
         thread_safe_environment = true;
     }
 
-    VMapManager2::~VMapManager2(void)
+    // using c++ 11 static initializer, we have the guarantee of creating only one MapManager instance (aka singleton)
+    VMapManager* VMapManager::createOrGetVMapManager()
+    {
+        static VMapManager gVMapManager;
+        return &gVMapManager;
+    }
```

Comment:
createOrGetXMapManager() returns a pointer. not a reference. To change that would require to modify the code of every calls to this method. I would like to avoid that.